clconf/clconf.go

```diff
@@ -336,38 +344,46 @@ func ToKvMap(conf interface{}) map[string]string {
 
 // UnmarshalSingleYaml will unmarshal a single yaml/json string without merging.
 // This form works for any yaml data, not just objects.
-func UnmarshalSingleYaml(yamlString string) (interface{}, error) {
-	var value interface{}
-	err := yaml.Unmarshal([]byte(yamlString), &value)
-	return value, err
-}
-
-func unmarshalYaml(yamlBytes ...[]byte) (map[interface{}]interface{}, error) {
-	result := make(map[interface{}]interface{})
-	for index := len(yamlBytes) - 1; index >= 0; index-- {
-		yamlMap := make(map[interface{}]interface{})
-
-		err := yaml.Unmarshal(yamlBytes[index], &yamlMap)
-		if err != nil {
-			return nil, err
+func UnmarshalSingleYaml(yamlString string) ([]interface{}, error) {
```

Comment:
This again breaks the contract (and the implied contract in the name)...  I think this should unmarshal the first yaml on the string (just a documentation update) and then we create a new function:

```go
func UnmarshalSingleYaml(yamlString string) (interface{}, error) {
    yamls, err := UnmarshalYamlString(yamlString string)
    if err != nil {
        return nil, err
    }
    return yamls[0], nil
}

func UnmarshalYamlString(yamlString string) ([]interface{}, error) {
   ...
}

Reply:
My question is why you would ever want to silently ignore data? If there are multiple files in the string shouldn't they be processed (or have an error thrown)? I think having it return an array forces one to realize that there could be more than one doc in the input.

Alternatively I _almost_ made `UnmarshalSingleYaml` do the merging of that yaml if there is more than one doc involved.

That said I don't feel too strongly about it, so I can definitely implement the suggestion above.

Reply:
I feel 100% strongly about _don't break the contract_.

Having `UnmarshalSingleYaml` do its own internal merge might work, but be careful with the merge order...  I would not be against this if you can make it work, but will have to make another helper to do the merge so it can by called here and in the other method that aggregates all of them.

Isn't silently ignore data what the underlying library does?  If you don't loop over the read it will just read to the end of the first doc and return.  In any case, the documentation for this method can clearly state that it will only read the first yaml file if multiple are in the string.  It clearly states that in the method name.

Reply:
[Reverted](https://github.com/pastdev/clconf/pull/16/commits/93a001f6441aa3013080a3d152dff92ae24058e8)